code/obj/item/device/audio_log.dm

```diff
@@ -68,103 +67,101 @@
 
 			return round((messages.len /  max_lines) * 100)
 
+#define MODE_OFF 0
+#define MODE_RECORDING 1
+#define MODE_PLAYING 2
 
 /obj/item/device/audio_log
 	name = "audio log"
 	desc = "A fairly spartan recording device."
 	icon_state = "recorder"
-	uses_multiple_icon_states = 1
+	uses_multiple_icon_states = TRUE
 	item_state = "electronic"
 	w_class = W_CLASS_SMALL
 	var/obj/item/audio_tape/tape = null
-	var/mode = 0 //1 recording, 2 playing back
+	var/mode = MODE_OFF
 	var/max_lines = 60
-	var/continuous = 1
+	var/text_colour = "#3FCC3F"
+	var/continuous = TRUE
+	var/list/name_colours = list()
 	var/list/audiolog_messages = list()
 	var/list/audiolog_speakers = list()
-	var/self_destruct = 0 //This message will self-destruct in five seconds...
+	var/self_destruct = FALSE
 	mats = 4
 
-	//nuclear mode briefing log
-	nuke_briefing
-		name = "Mission Briefing"
-		desc = "The standard for covert mission briefing."
-		continuous = 0
-		//self_destruct = 1
+	wall_mounted
+		name = "Mounted Logger"
+		desc = "A wall-mounted audio log device."
+		max_lines = 30
 
-		New(newloc, var/nuke_area)
-			..()
-			if(!nuke_area)
-				nuke_area = "an unknown area. I think mission control fucked up somewhere."
-			src.audiolog_messages += "Your mission this time is simple, team."
-			src.audiolog_messages += "NanoTrasen has been causing us significant trouble recently."
-			src.audiolog_messages += "You are to detonate their station with a nuclear device."
-			src.audiolog_messages += "You must arm the bomb in [nuke_area]. Good luck and god speed."
-			src.audiolog_speakers.len = length(src.audiolog_messages)
+		attack_hand(mob/user)
+			return attack_self(user)
+
+		updateSelfDialog()
+			return updateUsrDialog()
 
+	New()
+		..()
+		if (!src.chat_text)
+			src.chat_text = new
+		src.vis_contents += src.chat_text
+		SPAWN(1 SECOND)
 			if (!src.tape)
 				src.tape = new /obj/item/audio_tape(src)
+			if (src.audiolog_messages && length(src.audiolog_messages))
+				src.tape.messages = src.audiolog_messages
+				src.audiolog_messages = null
+			if (src.audiolog_speakers && length(src.audiolog_speakers))
+				src.tape.speakers = src.audiolog_speakers
+				src.audiolog_speakers = null
 
-			src.tape.messages = src.audiolog_messages
-			src.audiolog_messages = null
-
-			src.tape.speakers = src.audiolog_speakers
-			src.audiolog_speakers = null
-
+	Topic(href, href_list)
+		..()
+		if (usr.stat || usr.restrained() || usr.lying)
 			return
+		if ((usr.contents.Find(src) || usr.contents.Find(src.master) || in_interact_range(src, usr) && istype(src.loc, /turf)))
```

Comment:
don't use `Find` unless you need the index, use `in`

Reply:
From the original code, just moved, the same with `.len` for the most part. I'll amend them now.